.github/workflows/ci.yaml

```diff
@@ -190,15 +199,30 @@ jobs:
 
       - name: Set up JDK ${{ env.INT_TEST_JAVA_RUNTIME_VERSION }} (${{ env.INT_TEST_JAVA_RUNTIME_VENDOR }}) to run integration test
         uses: actions/setup-java@v4
-        if: ${{ !(env.INT_TEST_JAVA_RUNTIME_VERSION == env.JAVA_VERSION && env.INT_TEST_JAVA_RUNTIME_VENDOR == env.JAVA_VENDOR) }}
+        if: ${{ env.SET_UP_INT_TEST_RUNTIME_JDK_WHEN_NOT_ORACLE_8_OR_11 == 'true'}}
         with:
           java-version: ${{ env.INT_TEST_JAVA_RUNTIME_VERSION }}
           distribution: ${{ env.INT_TEST_JAVA_RUNTIME_VENDOR }}
 
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        if: ${{ env.SET_UP_INT_TEST_RUNTIME_ORACLE_JDK_8_OR_11 == 'true'}}
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.CR_PAT }}
+
+      - name: Set up JDK ${{ env.INT_TEST_JAVA_RUNTIME_VERSION }} (oracle) to run the integration test
+        if: ${{ env.SET_UP_INT_TEST_RUNTIME_ORACLE_JDK_8_OR_11 == 'true'}}
+        run: |
+          container_id=$(docker create "ghcr.io/scalar-labs/oracle/jdk:${{ env.INT_TEST_JAVA_RUNTIME_VERSION }}-linux")
```

Comment:
This is just a slight concern and I don't have answer at the moment, but it might seem a kind of redistribution of something. I think it's basically okay to use the distribution for testing purpose, but some consideration might be needed when it comes to redistribution?

Reply:
The [license](https://www.oracle.com/downloads/licenses/javase-license1.html) states: 
```
You may not [...] make the Programs available in any manner to any third party (other than Contractors acting on Your behalf as set forth in this Agreement);
```

I don't know if storing it on Github constitutes a third party. Is that your point?
I agree that getting legal advice might be a good idea.

Reply:
> Is that your point?

Basically, yes. I may be missing other points, though.

Sorry, I don't have enough knowledge to be sure it's legally okay or not. So, I'll continue review this PR from technical perspective.

@feeblefakie It would be great if you can ask someone who is a legal expert on this issue.

Reply:
@Torch3333 @komamitsu 
I think it's OK to use it for testing/development purposes as below.

```
License Rights and Restrictions Oracle grants You a nonexclusive, nontransferable, limited license to use the Programs, subject to the restrictions stated in this Agreement and Program Documentation, only for:

(i)     Personal Use,
(ii)    Development Use,
(iii)   Oracle Approved Product Use, and/or
(iv)   Oracle Cloud Infrastructure Use.
```

```
“Development Use” refers to Your internal use of the Programs to develop, test, prototype and demonstrate Your Applications. For purposes of clarity, the “to develop” grant includes using the Programs to run profilers, debuggers and Integrated Development Environments (IDE Tools) where the primary purpose of the IDE Tools is profiling, debugging and source code editing Applications.
```